spec/models/service_template_ansible_playbook_spec.rb

```diff
@@ -147,4 +161,47 @@
       expect(service_template.options[:config_info]).to include(saved_options)
     end
   end
+
+  describe '#update_catalog_item' do
+    it 'updates and returns the modified catalog item' do
+      service_template = prebuild_service_template
+      expect(described_class).to receive(:create_new_dialog)
+      expect(ManageIQ::Providers::EmbeddedAnsible::AutomationManager::ConfigurationScript).to receive(:update_in_provider_queue).once
+      service_template.update_catalog_item(catalog_item_options_three, user)
+
+      expect(service_template.name).to eq(catalog_item_options_three[:name])
+      expect(service_template.description).to eq(catalog_item_options_three[:description])
+      expect(service_template.options.fetch_path(:config_info, :provision, :extra_vars)).to have_attributes(
+        'key1' => 'updated_val1',
+        'key2' => 'updated_val2'
+      )
+      expect(service_template.dialogs.first.name).to eq catalog_item_options_three
+        .fetch_path(:config_info, :provision, :new_dialog_name)
+    end
```

Comment:
Need to add `expect(service_template.resource_actions.first)` to contain newly created/updated job template.